corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/entities/menu.js

```diff
@@ -0,0 +1,126 @@
+
+        parse: function (response) {
+            this.sessionId = response.session_id;
+            this.sequenceId = response.seq_id;
+            this.title = response.title;
+            FormplayerFrontend.request('currentUser').sessionId = this.sessionId;
+
+            if (response.commands) {
+                this.type = "commands";
+                return response.commands;
+            }
+            else if (response.entities) {
+                this.type = "entities";
+                this.action = response.action;
+                this.styles = response.styles;
+                return response.entities;
+            }
+            else if (response.details) {
+                this.type = "details";
+                this.styles = response.styles;
+                this.headers = response.headers;
+                this.details = response.details;
+
+                var model = [];
+                for (i = 0; i < this.details.length; i += 1) {
+                    current = {'header': this.headers[i], 'data': this.details[i]};
+                    model.push(current)
+                }
+                return model;
+            }
+        },
+
+        initialize: function (params) {
+            this.domain = params.domain;
+            this.app_id = params.app_id;
+            this.fetch = params.fetch;
+            this.sessionId = params.sessionId;
+            this.sequenceId = params.sequenceId;
+            this.selection = params.selection;
+        }
+    });
+
+    var API = {
+
+        getDetail: function (model) {
+
+            var defer = $.Deferred();
+            var mCollection = new Backbone.Collection.extend({
+                fetch: function(){
+                    return model.options.model.attributes.detail
+                },
+                initialize: function (params) {
+                    this.fetch = params.fetch;
+                }
+            });
+            mCollection.fetch({
+                success: function(request) {
+                    deger.resolve(request);
+                }
+            });
+            return defer.promise();
+        },
+
+        getMenus: function (app_id, select_list) {
+            var menus = new Entities.MenuSelectCollection({
+                domain: FormplayerFrontend.request('currentUser').domain,
+                app_id: app_id,
+
+                fetch: function (options) {
+                    var collection = this;
+
+                    options.data = JSON.stringify({
+                        "username": "test",
+                        "password": "123",
+                        "domain": "test",
+                        "app_id": collection.app_id,
+                        "selections": select_list
+                    });
+
+                    if (select_list) {
+                        options.data.selections = select_list;
+                    }
+
+                    options.url = 'http://localhost:8080/navigate_menu';
```

Comment:
we'll still need to hit the django server first as a proxy because we can't make cross site requests